lib/libtsd/tsd_task.c

```diff
@@ -326,10 +326,14 @@ tsd_task_start(struct tsd_task *t)
 			if (ret != 0)
 				_exit(1);
 		}
-		if (getgid() != getegid())
-			(void)setgid(getgid());
-		if (getuid() != geteuid())
-			(void)setuid(getuid());
+		if (getgid() != getegid() && setgid(getgid()) != 0) {
+			ERROR("failed to reset process group");
+			goto fail;
+		}
+		else if (getuid() != geteuid() && setuid(getuid()) != 0) {
```

Comment:
Pull up as per [style(9)](https://www.freebsd.org/cgi/man.cgi?query=style&sektion=9), or just get rid of the braces altogether, as they are unnecessary once you remove the incorrect `goto fail`.